src/lib/journey/callbacks/recaptcha/recaptcha.svelte

```diff
@@ -23,57 +38,59 @@
     callback?.getOutputByName<string>('captchaDivClass', 'h-captcha') ?? 'h-captcha';
 
   onMount(() => {
-    window.frHandleCaptchaError = function handleCaptchaError() {
-      callback?.setResult('');
-      if (recaptchaClass.startsWith('h')) {
-        window.hcaptcha.render(recaptchaClass, {
-          sitekey: siteKey,
-          callback: 'frHandleCaptcha',
-          'expired-callback': 'frHandleExpiredCallback',
-          'chalexpired-callback': 'frHandleExpiredCallback',
-          'error-callback': 'frHandleErrorCallback',
-        });
-      } else {
-        window.grecaptcha.render(recaptchaClass, {
-          sitekey: siteKey,
-          callback: window.frHandleCaptcha, // grecaptcha uses different types so passing real function here
-          'expired-callback': window.frHandleExpiredCallback,
-        });
-      }
-    };
-    window.frHandleCaptcha = function handleCaptchaToken(token: string) {
-      callback?.setResult(token);
-    };
-    window.frHandleExpiredCallback = function handleExpiredCallback() {
-      callback?.setResult('');
-      if (recaptchaClass.startsWith('h')) {
-        window.hcaptcha.render(recaptchaClass, {
-          sitekey: siteKey,
-          callback: 'frHandleCaptcha',
-          'expired-callback': 'frHandleExpiredCallback',
-          'chalexpired-callback': 'frHandleExpiredCallback',
-          'error-callback': 'frHandleErrorCallback',
-        });
-      } else {
-        window.grecaptcha.render(recaptchaClass, {
-          sitekey: siteKey,
-          callback: window.frHandleCaptcha, // grecaptcha uses different types so passing real function here
-          'expired-callback': window.frHandleExpiredCallback,
+    if (isV3) {
+      // If ReCaptcha v3, do nothing and return early
+      return;
+    }
+    if (callback) {
+      window.frHandleCaptchaError = handleCaptchaError(callback);
+      window.frHandleCaptcha = handleCaptchaToken(callback);
+      window.frHandleExpiredCallback = function handleExpiredCallback() {
+        callback?.setResult('');
+        renderCaptcha({ nameOfCaptcha: 'hcaptcha', siteKey });
+      };
+      renderCaptcha({
+        nameOfCaptcha: recaptchaClass === 'g-recaptcha' ? 'grecaptcha' : 'hcaptcha',
+        siteKey,
+      });
+    }
+  });
+  // defining this outside of the reactive block and guarding it with a isV3 check so it only runs when v3
+  // is defined as true and we have a recaptcha action to assign.
+  function executeV3Captcha() {
+    if (isV3 && recaptchaAction.length) {
```

Comment:
Is an action value a required attribute for v3? We provide a default value for `recaptchaAction`, yes? If so, would this ever be an empty string?

Reply:
Okay, I think I get it now, and you're right, there are instances where we don't have a journey, and maybe just a `suspendedId` or something. Good thinking.